sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -267,6 +266,17 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
                 _currentActivity = StartActivitySourceActivity();
                 if (_currentActivity != null)
                 {
+                    _isAllDataRequested = _currentActivity.GetIsAllDataRequestedFlag();
+
+                    if (!_isAllDataRequested)
+                    {
+                        _sampleOutActivity = _currentActivity;
+                        _currentActivity.Stop();
```

Comment:
We shouldn't stop the current activity.

Reply:
Are tests passing currently? Ideally we would have a test that would fail with this change.

Reply:
yes..client diagnostics tests are passing with current change